build/Makefile

```diff
@@ -0,0 +1,169 @@
+SHELL:=/usr/bin/env bash
+SRC_DIR:=../src
+SRC_FILES:=$(shell find $(SRC_DIR) -name '*.xml' -o -name '*.json')
+SRC_XML_PROFILES:=$(shell find $(SRC_DIR) -name '*profile.xml')
+SRC_READMES:=$(shell find $(SRC_DIR) -iname 'README.md')
+GEN_CONTENT_DIR:=generated
+GEN_READMES:=$(patsubst $(SRC_DIR)/%,$(GEN_CONTENT_DIR)/%,$(SRC_READMES))
+GEN_XML_FILES:=$(patsubst $(SRC_DIR)/%,$(GEN_CONTENT_DIR)/%,$(SRC_FILES))
+GEN_XML_PROFILES:=$(subst _profile.xml,-resolved-profile_catalog.xml,$(patsubst $(SRC_DIR)/%,$(GEN_CONTENT_DIR)/%,$(SRC_XML_PROFILES)))
+GEN_JSON_FILES:=$(subst xml,json,$(GEN_XML_FILES))
+GEN_MIN_JSON_FILES:=$(subst .json,-min.json,$(subst xml,json,$(GEN_XML_FILES)))
+GEN_YAML_FILES:=$(subst xml,yaml,$(GEN_XML_FILES))
+CURL_INSTALL_OPTS:=--silent --location
+JQ_BIN:=jq-linux-amd64
+JQ_PATH:=$(shell pwd)/jq
+JQ_INSTALL_VERSION:=jq-1.7
+JQ_DOWNLOAD_URL:=https://github.com/jqlang/jq/releases/download/$(JQ_INSTALL_VERSION)/$(JQ_BIN)
+JQ_INSTALL_COMMAND=curl $(CURL_INSTALL_OPTS) -o $(JQ_PATH) $(JQ_DOWNLOAD_URL) && chmod +x $(JQ_PATH)
+XMLLINT_PATH=/usr/bin/xmllint
+XMLLINT_INSTALL_COMMAND=sudo rm -rf /var/lib/apt/lists/* && sudo apt-get update && sudo apt-get install libxml2-utils
+YQ_BIN:=yq_linux_amd64
+YQ_PATH:=$(shell pwd)/yq
+YQ_INSTALL_VERSION:=v4.35.2
+YQ_DOWNLOAD_URL:=https://github.com/mikefarah/yq/releases/download/$(YQ_INSTALL_VERSION)/$(YQ_BIN)
+YQ_INSTALL_COMMAND:=curl $(CURL_INSTALL_OPTS) -o $(YQ_PATH) $(YQ_DOWNLOAD_URL) && chmod +x $(YQ_PATH)
```

Comment:
B: I like this approach, however it unfortunately breaks in non-linux environments. The XML lint install also breaks on non-debian/ubuntu environments. Maybe we should just document these requirements in the readme as was done in the core oscal repo?

Reply:
I thought the way you fix that for other Linux/Unixy environments it override the command on the command line, I actually tested this and was surprised how effective it is, but this means less, how can I say this, "elaborate" arguments when invoking `make` in GitHub Actions if we need to express them explicitly there, so I though it wasn't worth the headache.

Conversely, if you explain with the `XMLLINT_PATH` absolute target to where that file is, you can override to wherever you want. The only place I cannot see this work is Windows, am I mistaken?

Reply:
Done in https://github.com/usnistgov/oscal-content/pull/204/commits/49a77648618e827838540385085228f90297ba44.